sdk/identity/Azure.Identity/src/ImdsManagedIdentitySource.cs

```diff
@@ -72,13 +70,16 @@ protected override Request CreateRequest(string[] scopes)
 
             request.Uri.AppendQuery("resource", resource);
 
-            if (!string.IsNullOrEmpty(_clientId))
+            string idQueryParam = _managedIdentityId?._idType switch
             {
-                request.Uri.AppendQuery(Constants.ManagedIdentityClientId, _clientId);
-            }
-            if (!string.IsNullOrEmpty(_resourceId))
+                ManagedIdentityIdType.ClientId => Constants.ManagedIdentityClientId,
+                ManagedIdentityIdType.ResourceId => Constants.ManagedIdentityResourceId,
```

Comment:
Should this handle `ObjectId`?

Reply:
This source is currently unused, since we migrated to the MSAL implementation - it's only been changed to accommodate the options type change.